README.md

```diff
@@ -218,6 +223,50 @@ To start with the actual implementation, you may check our [IntelliJ Platform SD
 For those, who value example codes the most, there are also available [IntelliJ SDK Code Samples][gh:code-samples] and [IntelliJ Platform Explorer][jb:ipe] – a search tool for browsing Extension Points inside existing implementations of open-source IntelliJ Platform plugins.
 
 
+## Testing
+
+[Testing plugins][docs:testing-plugins] is an essential part of the plugin development for the IntelliJ-based IDEs to make sure that everything works as expected between IDE releases and plugin refactorings.
+The IntelliJ Platform Plugin Template project provides examples of two testing approaches – functional and UI tests.
+
+### Functional tests
+
+Most of the IntelliJ Platform codebase tests are model-level, run in a headless environment using an actual IDE instance.
+The tests usually test a feature as a whole rather than individual functions that comprise its implementation, like in unit tests.
+
+Depending on the feature type and its execution time, there are two types of tests available in the IntelliJ Platform test framework:
+- heavy tests – each test runs in a new project,
+- light tests – all tests share the same project for running.
+
+Check the [Light and Heavy Tests][docs:light-and-heavy-tests] documentation section for more information.
+
+In `src/test/kotlin`, you'll find a basic test that utilizes `BasePlatformTestCase` and runs a few checks against the XML files to indicate an example operation of creating files on the fly or reading them from `src/test/resources/rename` test resources.
+
+> **TIP:** Run your tests using predefined *Run Tests* configuration or by invoking the `./gradlew test` Gradle task.
+
+### UI tests
+
+If your plugin provides custom user interfaces, you should consider covering them with tests and the functionality they utilize.
```

Comment:
custom user interface ->
complex user interace

there is usually no need to build UI tests for simple things like custom settings page using IJ default Swing components